src/async_op.c

```diff
@@ -35,30 +41,69 @@ static void async_op_dispose(ASYNC_OP* async_op)
 #define IS_POWER_OF_2(x) \
     (((x) & ((x) - 1)) == 0)
 
+/*tiny function that returns the next multiple of "align" starting from "start" (included). Does 1 mod and 1 branch*/
+static void* get_next_aligned_address(void* start, uint32_t align)
+{
+    void* result;
+    uint32_t m = (uintptr_t)start % align;
+    if (m == 0)
+    {
+        result = start;
+    }
+    else
+    {
+        result = (void*)((uintptr_t)start - m + align);
+    }
+    return result;
+}
+
+
+static void* get_prev_aligned_with_size_address(void* start, uint32_t size, uint32_t align)
+{
+    uintptr_t first_prev_address = (uintptr_t)start - size;
+    uint32_t m = first_prev_address % align;
+    if (m == 0)
+    {
+        return (void*)first_prev_address;
+    }
+    else
+    {
+        return (void*)((uintptr_t)first_prev_address - align);
+    }
+}
+
+MU_STATIC_ASSERT(UINT32_MAX - 1 - 1 > sizeof(CONTEXT_TO_ASYNC_OP));

```

Comment:
>- 1 - 1 [](http://example.com/codeflow?start=28&length=7)

Won't you like have -2 instead? or some kind of constant like #define UINT32_ALIGN_OFFSET 2

Reply:
I chose to use `-1 -1` to be more explicit about the origins or each element (.md file has references). A wild '-2' would have raised more questions, imho.